### PR TITLE
Make grammar cross-referenced (generated)

### PR DESCRIPTION
**Pull request by nikic**

### CODE REVIEW

**Comment by nikic** [comment]
Examples: https://github.com/nikic/php-langspec/blob/extractGrammar/spec/10-expressions.md#grammar-callable-variable (in spec) https://github.com/nikic/php-langspec/blob/extractGrammar/spec/19-grammar.md#grammar-callable-variable (in grammar listing)

This replaces the "defined elsewhere" blocks.

**Comment by nikic** [comment]
Merged.

File: spec/09-lexical-structure.md [hunk 1]
```diff
@@ -43,25 +43,38 @@ successive indented line contains a possible expansion of the
...
+  'a' 'b' 'c' 'd' 'e' 'f'
+  'A' 'B' 'C' 'D' 'E' 'F'
+-->
+
 <pre>
-  <i>hexadecimal-digit:: one of</i>
-    0   1   2   3   4   5   6   7   8   9
-    a   b   c   d   e   f
-    A   B   C   D   E   F
+<a name="grammar-hexadecimal-digit-example"><i>hexadecimal-digit-example:: one of</i>
```
Comment (Hywan) [inline spec/09-lexical-structure.md]: The `<a`> element does not have a closing one.

File: spec/09-lexical-structure.md [hunk 2]
```diff
@@ -43,25 +43,38 @@ successive indented line contains a possible expansion of the
...
+single-line-comment-example::
+  '//' input-characters?
+  '#' input-characters?
+-->
+
 <pre>
-  <i>single-line-comment::</i>
-    // input-characters<sub>opt</sub>
-    #  input-characters<sub>opt</sub>
+<a name="grammar-single-line-comment-example"><i>single-line-comment-example::</i>
```
Comment (Hywan) [inline spec/09-lexical-structure.md]: The `<a>` element does not have a closing one.

File: spec/07-variables.md [hunk 3]
```diff
@@ -350,19 +360,23 @@ echo "\$fs = $fs\n";    // $fs = 3
...
+-->
 
-**Defined elsewhere**
+<pre>
+<a name="grammar-global-declaration"><i>global-declaration:</i>
+   global   <i><a href="#grammar-variable-name-list">variable-name-list</a></i>   ;
 
-* [*expression*](10-expressions.md#general-6)
-* [*simple-variable*](10-expressions.md#simple-variable)
+<a name="grammar-variable-name-list"><i>variable-name-list:</i>
```
Comment (Hywan) [inline spec/07-variables.md]: The `<a>` element does not have a closing one.

File: spec/07-variables.md [hunk 4]
```diff
@@ -266,25 +266,35 @@ $b = &colors[100];      // a VSlot for $b is created which points to the array
...
+   <i><a href="#grammar-static-variable-declaration">static-variable-declaration</a></i>
+   <i><a href="#grammar-static-variable-name-list">static-variable-name-list</a></i>   ,   <i><a href="#grammar-static-variable-declaration">static-variable-declaration</a></i>
 
-**Defined elsewhere**
+<a name="grammar-static-variable-declaration"><i>static-variable-declaration:</i>
+   <i><a href="09-lexical-structure.md#grammar-variable-name">variable-name</a></i>   <i><a href="#grammar-function-static-initializer">function-static-initializer</a></i><sub>opt</sub>
 
-* [*variable-name*](09-lexical-structure.md#names)
-* [*constant-expression*](10-expressions.md#constant-expressions)
+<a name="grammar-function-static-initializer"><i>function-static-initializer:</i>
```
Comment (Hywan) [inline spec/07-variables.md]: The `<a>` element does not have a closing one.

File: spec/07-variables.md [hunk 5]
```diff
@@ -266,25 +266,35 @@ $b = &colors[100];      // a VSlot for $b is created which points to the array
...
-</pre>
+function-static-initializer:
+  '=' constant-expression
+-->
+
+<pre>
+<a name="grammar-function-static-declaration"><i>function-static-declaration:</i>
+   static   <i><a href="#grammar-static-variable-name-list">static-variable-name-list</a></i>   ;
+
+<a name="grammar-static-variable-name-list"><i>static-variable-name-list:</i>
```
Comment (Hywan) [inline spec/07-variables.md]: The `<a>` element does not have a closing one.

File: spec/07-variables.md [hunk 6]
```diff
@@ -266,25 +266,35 @@ $b = &colors[100];      // a VSlot for $b is created which points to the array
...
 
-  <i>function-static-initializer:</i>
-    = <i>constant-expression</i>
-</pre>
+function-static-initializer:
+  '=' constant-expression
+-->
+
+<pre>
+<a name="grammar-function-static-declaration"><i>function-static-declaration:</i>
```
Comment (Hywan) [inline spec/07-variables.md]: The `<a>` element does not have a closing one.

File: spec/09-lexical-structure.md [hunk 7]
```diff
@@ -73,21 +86,27 @@ script. Each script must conform to this production.
...
+  comment
+  white-space
+  token
+-->
 
-* [*comment*](#comments)
-* [*white-space*](#white-space)
-* [*token*](#tokens)
+<pre>
+<a name="grammar-input-file"><i>input-file::</i>
```
Comment (Hywan) [inline spec/09-lexical-structure.md]: etc.

**Comment by nikic** [comment]
@Hywan Thanks! I've change it to use an `id` on the `<i>` tag in https://github.com/nikic/php-langspec/commit/ae3a7639a05fae1cd1690ef28213bfebcb7d3af0 instead. It seems that as of HTML5 using `<a name="">` is deprecated.

**Comment by Hywan** [comment]
Correct, this is deprecated since a long time, but still widely supported.

**Pull request closed without merging**